compiler/AST/TryStmt.cpp

```diff
@@ -52,30 +70,61 @@ void TryStmt::accept(AstVisitor* visitor) {
     if (_body) {
       _body->accept(visitor);
     }
+
+    for_alist(c, _catches) {
+      CatchStmt* catchStmt = toCatchStmt(c);
+      catchStmt->accept(visitor);
+    }
+
     visitor->exitTryStmt(this);
   }
 }
 
-Expr* TryStmt::copy(SymbolMap* map, bool internal) {
-  return NULL;
-}
-
-Expr* TryStmt::copyInner(SymbolMap* map) {
-  return NULL;
+TryStmt* TryStmt::copyInner(SymbolMap* map) {
+  TryStmt* copy = new TryStmt(_tryBang, COPY_INT(_body), NULL);
+  for_alist(c, _catches) {
+    copy->_catches.insertAtTail(c->copy());
+  }
+  return copy;
 }
 
 void TryStmt::replaceChild(Expr* old_ast, Expr* new_ast) {
-
+  if (old_ast == _body) {
+    _body = toBlockStmt(new_ast);
+  } else if (old_ast->list == &_catches) {
```

Comment:
replaceChild on an AList member works automatically. It uses the member's prev/next/parent pointers. The member's/AList's enclosing class should not do anything about its ALists in its replaceChild.
So, what Michael said.